plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/bootstrap/ServerConfig.java

```diff
@@ -0,0 +1,227 @@
+    }
+
+    /**
+     * Gets the thread pool executor builder configured for the Flight server.
+     *
+     * @return The configured ScalingExecutorBuilder instance
+     */
+    public static ScalingExecutorBuilder getClientExecutorBuilder() {
+        return new ScalingExecutorBuilder(FLIGHT_CLIENT_THREAD_POOL_NAME, threadPoolMin, threadPoolMax, keepAlive);
+    }
+
+    /**
+     * Returns a list of all settings managed by this configuration class.
+     *
+     * @return List of Setting instances
+     */
+    public static List<Setting<?>> getSettings() {
+        return new ArrayList<>(
+            Arrays.asList(
+                ARROW_ALLOCATION_MANAGER_TYPE,
+                ARROW_ENABLE_NULL_CHECK_FOR_GET,
+                ARROW_ENABLE_DEBUG_ALLOCATOR,
+                ARROW_ENABLE_UNSAFE_MEMORY_ACCESS,
+                ARROW_SSL_ENABLE
+            )
+        ) {
+            {
+                addAll(Netty4Configs.getSettings());
+            }
+        };
+    }
+
+    static EventLoopGroup createELG(String name, int eventLoopThreads) {
+
+        return Epoll.isAvailable()
+            ? new EpollEventLoopGroup(eventLoopThreads, new DefaultThreadFactory(name, true))
+            : new NioEventLoopGroup(eventLoopThreads, new DefaultThreadFactory(name, true));
+    }
+
+    static Class<? extends Channel> serverChannelType() {
+        return Epoll.isAvailable() ? EpollServerSocketChannel.class : NioServerSocketChannel.class;
+    }
+
+    static Class<? extends Channel> clientChannelType() {
+        return Epoll.isAvailable() ? EpollSocketChannel.class : NioSocketChannel.class;
+    }
+
+    private static class Netty4Configs {
+        public static final Setting<Integer> NETTY_ALLOCATOR_NUM_DIRECT_ARENAS = Setting.intSetting(
+            "io.netty.allocator.numDirectArenas",
+            1, // TODO - 2 * the number of available processors; to be confirmed and set after running benchmarks
+            1,
+            Setting.Property.NodeScope
+        );
+
+        public static final Setting<Boolean> NETTY_TRY_REFLECTION_SET_ACCESSIBLE = Setting.boolSetting(
+            "io.netty.tryReflectionSetAccessible",
+            true,
+            Setting.Property.NodeScope
+        );
+
+        public static final Setting<Boolean> NETTY_NO_UNSAFE = Setting.boolSetting("io.netty.noUnsafe", false, Setting.Property.NodeScope);
+
+        public static final Setting<Boolean> NETTY_TRY_UNSAFE = Setting.boolSetting("io.netty.tryUnsafe", true, Setting.Property.NodeScope);
+
+        @SuppressForbidden(reason = "required for netty allocator configuration")
+        public static void init(Settings settings) {
+            System.setProperty("io.netty.allocator.numDirectArenas", Integer.toString(NETTY_ALLOCATOR_NUM_DIRECT_ARENAS.get(settings)));
+            System.setProperty("io.netty.noUnsafe", Boolean.toString(NETTY_NO_UNSAFE.get(settings)));
```

Comment:
We cannot do that: netty is used in default transports other plugins, the attempts to modify these system props will affect all of them

Reply:
@reta since each plugin have a separate classloader, so each will have it own version of these properties. However, if `netty-common` is part of app classloader, then what you're saying is correct. Let me know if my understanding is correct. Setting these sys properties will initialize static classes of netty present in this plugin's classpath.

Reply:
`System.setProperty()` are not set per classloader

Reply:
@reta what do you suggest here?

Reply:
If we could remove those for now, it would be a best option

Reply:
this will break RootAllocator initialization

Reply:
related issue - https://github.com/apache/arrow-java/issues/123 for `io.netty.noUnsafe`.

Reply:
Confused now - we define setting for it, but the only value that works is actually `false` (in case of `noUnsafe`)?  I think we should get rid of settings, those are not working anyway, right? 

So how could we solve it, I have 2 options in mind:
 - on plugin initialization, if FeatureFlags (for Arrow Flight) is set, check if these system properties are present, if not - ask user to add them to `jvm.options` or `OPENSEARCH_JAVA_OPTS` explicitily
 - in the server's Bootstrap class, check , if FeatureFlags (for Arrow Flight) is set, than set these system properties before initialization happens 

Anything else?

Reply:
@reta 
Can we set and reset them back to their original value per auxiliary transport? 
If we ensure each auxiliary transport initialization happens sequentially, then this approach might work. 

Another option could be patching netty in flight-rpc plugin to read it from a different custom environment variable?

I kept them as settings as its possible to switch to non-netty based allocator when using `arrow-memory-core`, but given we are always defaulting to netty based allocator in OpenSearch, this setting doesn't makes sense.

Reply:
> Can we set and reset them back to their original value per auxiliary transport?

We do not have control over initialization sequence, it its not about aux transport, it involves modules (transport-netty4), plugins (security, repository-s3, transport-reactor-netty4) and aux transport plugins (grpc transport)

> Another option could be patching netty in flight-rpc plugin to read it from a different custom environment variable?

I (personally) prefer not doing that. It is fine to ask or set this system properties if we need to, but doing it consistently would eliminate any weird initialization flows (depending what comes first).

Reply:
sounds good! right now we are not initializing StreamManagerPlugin, it will happen in next PR.
For now, I have removed them.